### PR TITLE
Add migration guide

### PR DESCRIPTION
**Pull request by kpob**

### CODE REVIEW

File: docusaurus/docs/migrations/to-0.8.0.md [hunk 1]
```diff
@@ -3,71 +3,1090 @@ sidebar_position: 1
...
+
+### 2.1 **Add bin directory**
+Odra 0.8.0 introduces a new way to build smart contracts. The `.builder_casper` directory is no longer used. Instead, you should create a new directory called `bin` in the root of your project and add the `build_contract.rs` and `build_schema.rs` files to the `bin` directory.
+
+You can find the `build_contract.rs` and `build_schema.rs` files in [templates] directory in the Odra main repository. You can choose whatever template you want to use and copy the files to your project. In both files, you should replace `{{project-name}}` with the name of your project.
+
+
+### 2.2. **Update Cargo.toml**
+There a bunch of changes in the `Cargo.toml` file.
+* You don't to specify the features anymore - remove the `features` section and `default-features` flag from the `odra` dependency.
```
Comment (zie1ony) [inline docusaurus/docs/migrations/to-0.8.0.md:42]:
```suggestion
* You don't have to specify the features anymore - remove the `features` section and `default-features` flag from the `odra` dependency.
```

File: docusaurus/docs/migrations/to-0.8.0.md [hunk 2]
```diff
@@ -3,71 +3,1090 @@ sidebar_position: 1
...
+   * `Variable` is now `Var`.
+   * Remove `odra::contract_env`.
+   * Remove `odra::types::event::OdraEvent`.
+   * Remove `odra::types::OdraType` as it is no longer required.
+   * Change `odra::types::casper_types::*;` to `odra::casper_types::*;`.
+  
+#### 2.3.2. **Some type aliases are no longer in use.**
+   * `Balance` - use `odra::casper_types::U512`.
+   * `BlockTime` - use `u64`.
+   * `Event` - use `odra::casper_types::bytesrepr::Bytes`.
```
Comment (zie1ony) [inline docusaurus/docs/migrations/to-0.8.0.md:143]: `Event` as a macro?

File: docusaurus/docs/migrations/to-0.8.0.md [hunk 3]
```diff
@@ -3,71 +3,1090 @@ sidebar_position: 1
...
+
+#### 2.3.3. **Consider import `odra::prelude::*` in your module files.**
+
```
Comment (zie1ony) [inline docusaurus/docs/migrations/to-0.8.0.md:146]:
```suggestion
```rust
use odra::prelude::*;
```
```

File: docusaurus/docs/migrations/to-0.8.0.md [hunk 4]
```diff
@@ -3,71 +3,1090 @@ sidebar_position: 1
...
+    ownable: Ownable
+}
+```
+
+</TabItem>
+</Tabs>
+
+#### 2.3.9. **Update external contract calls.**
+
+However the definition of an external contract remains the same, the way you call it has changed. A reference to an external contract is named `{{ModuleName}}ContractRef` (former `{{ModuleName}}Ref`) and you can call it using `{{ModuleName}}ContractRef::new(self.env(), address)` (former `{{ModuleName}}Ref::at()`).
```
Comment (zie1ony) [inline docusaurus/docs/migrations/to-0.8.0.md:292]:
```suggestion
However the definition of an external contract remains the same, the way you call it has changed. A reference to an external contract is named `{{ModuleName}}ContractRef` (former `{{ModuleName}}Ref`) and you can call it using `{{ModuleName}}ContractRef::new(env, address)` (former `{{ModuleName}}Ref::at()`).
```

File: docusaurus/docs/migrations/to-0.8.0.md [hunk 5]
```diff
@@ -3,71 +3,1090 @@ sidebar_position: 1
...
+<TabItem value="current" label="0.8.0">
+
+```rust
+#[odra::external_contract]
+pub trait Token {
+    fn balance_of(&self, owner: &Address) -> U256;
+}
+
+// Usage
+TokenContractRef::new(self.env(), token).balance_of(account)
```
Comment (zie1ony) [inline docusaurus/docs/migrations/to-0.8.0.md:304]:
```suggestion
TokenContractRef::new(env, token).balance_of(account)
```

**Review by zie1ony — APPROVED** [review]
Awesome :)

**Review by kubaplas — APPROVED** [review]
:joy_cat:

File: docusaurus/docs/migrations/to-0.8.0.md (hunk 2 above)
Comment (kpob) [reply docusaurus/docs/migrations/to-0.8.0.md:143]: Should be `EventData`

**Pull request merged**